server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -50,41 +50,58 @@ def _get_default_config():
             "diffexp_lfc_cutoff": 0.01,
         }
 
-    def _alias_annotation_names(self, axis, name):
+    @staticmethod
+    def _create_unique_column_name(df, prefix):
+        """ given a dataframe and a name prefix, return a column name which does not
+            exist in the dataframe.
         """
-        Do all user-specified annotation aliasing.
+        suffix = 0
+        while f"{prefix}{suffix}" in df.columns:
+            suffix += 1
+        return f"{prefix}{suffix}"
 
-        As a *critical* side-effect, ensure the indices are simple number ranges
-        (accomplished by calling pandas.DataFrame.reset_index())
+    def _alias_annotation_names(self):
+        """
+        The front-end relies on the existance of a unique, human-readable
+        index for obs & var (eg, var is typically gene name, obs the cell name).
+        The user can specify these via the --obs-names and --var-names config.
+        If they are not specified, use the existing index to create them, giving
+        the resulting column a unique name (eg, "name").
+
+        In both cases, enforce that the result is unique, and communicate the
+        index column name to the front-end via the obs_names and var_names config
+        (which is incorporated into the schema).
         """
-        if name == "name":
-            # a noop, so skip it
-            return
-
-        ax_name = str(axis)
-        df_axis = getattr(self.data, ax_name)
-        if name is None:
-            # reset index to simple range; alias "name" to point at the
-            # previously specified index.
-            df_axis.reset_index(inplace=True)
-            df_axis.rename(inplace=True, columns={"index": "name"})
-        elif name in df_axis.columns:
-            if name not in df_axis.columns:
```

Comment:
I'm a little worried about doing git blame on the above two lines 😳

Reply:
you and me both :-).    I suspect I am likely guilty!!!